src/language/walker/ruleWalker.ts

```diff
@@ -65,7 +65,7 @@ export class RuleWalker extends SyntaxWalker implements IWalker {
     }
 
     /** @deprecated Prefer `addFailureAt` and its variants. */
-    public createFailure(start: number, width: number, failure: string, fix?: Fix): RuleFailure {
+    public createFailure(start: number, width: number, failure: string, fix?: Replacement | Replacement[]): RuleFailure {
```

Comment:
I tried that at first, but that can make life difficult if you have a function returning optional fixes. You would need to use `const fix = optionalFix(); ctx.addFailureAtNode(node, failure, ...(fix ? fix : []))`, instead of just `ctx.addFailureAtNode(node, failure, optionalFix())`.